src/Shared/ServerInfrastructure/BufferExtensions.cs

```diff
@@ -129,47 +129,43 @@ internal static void WriteAscii(ref this BufferWriter<PipeWriter> buffer, string
     }
 
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    internal static unsafe void WriteNumeric(ref this BufferWriter<PipeWriter> buffer, ulong number)
+    internal static void WriteNumeric(ref this BufferWriter<PipeWriter> buffer, ulong number)
     {
         const byte AsciiDigitStart = (byte)'0';
 
-        var span = buffer.Span;
-        var bytesLeftInBlock = span.Length;
+        var start = buffer.Span;
+        var bytesLeftInBlock = start.Length;
 
         // Fast path, try copying to the available memory directly
         var simpleWrite = true;
-        fixed (byte* output = span)
+        if (number < 10 && bytesLeftInBlock >= 1)
         {
-            var start = output;
-            if (number < 10 && bytesLeftInBlock >= 1)
-            {
-                *(start) = (byte)(((uint)number) + AsciiDigitStart);
-                buffer.Advance(1);
-            }
-            else if (number < 100 && bytesLeftInBlock >= 2)
-            {
-                var val = (uint)number;
-                var tens = (byte)((val * 205u) >> 11); // div10, valid to 1028
+            start[0] = (byte)(((uint)number) + AsciiDigitStart);
+            buffer.Advance(1);
+        }
+        else if (number < 100 && bytesLeftInBlock >= 2)
```

Comment:
What does it look like if `start.Length >= 2` is used?

Can the JIT then elide the unwanted code? IMO the JIT doesn't see the connection of `bytesLeftInBlock` and the length of the span, so we should avoid it.

Reply:
Answered in https://github.com/dotnet/aspnetcore/pull/57417#issuecomment-2301704775

Reply:
It does not make a difference, but updated as suggested.